azurerm/internal/services/network/firewall_policy_resource.go

```diff
@@ -0,0 +1,200 @@
+		FirewallPolicyPropertiesFormat: &network.FirewallPolicyPropertiesFormat{
+			ThreatIntelMode: network.AzureFirewallThreatIntelMode(d.Get("threat_intelligence_mode").(string)),
+		},
+		Location: utils.String(location.Normalize(d.Get("location").(string))),
+		Tags:     tags.Expand(d.Get("tags").(map[string]interface{})),
+	}
+	if id, ok := d.GetOk("base_policy_id"); ok {
+		param.FirewallPolicyPropertiesFormat.BasePolicy = &network.SubResource{ID: utils.String(id.(string))}
+	}
+
+	if _, err := client.CreateOrUpdate(ctx, resourceGroup, name, param); err != nil {
+		return fmt.Errorf("creating Firewall Policy %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	resp, err := client.Get(ctx, resourceGroup, name, "")
+	if err != nil {
+		return fmt.Errorf("retrieving Firewall Policy %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+	if resp.ID == nil || *resp.ID == "" {
+		return fmt.Errorf("empty or nil ID returned for Firewall Policy %q (Resource Group %q) ID", name, resourceGroup)
+	}
+	d.SetId(*resp.ID)
+
+	return resourceArmFirewallPolicyRead(d, meta)
+}
+
+func resourceArmFirewallPolicyRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Network.FirewallPoliciesClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.FirewallPolicyID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Get(ctx, id.ResourceGroup, id.Name, "")
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[DEBUG] Firewall Policy %q was not found in Resource Group %q - removing from state!", id.Name, id.ResourceGroup)
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("retrieving Firewall Policy %q (Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
+	}
+
+	d.Set("name", id.Name)
+	d.Set("resource_group_name", id.ResourceGroup)
+	d.Set("location", location.NormalizeNilable(resp.Location))
+
+	basePolicyID := ""
+	if resp.BasePolicy != nil && resp.BasePolicy.ID != nil {
+		basePolicyID = *resp.BasePolicy.ID
+	}
+	d.Set("base_policy_id", basePolicyID)
+
+	d.Set("threat_intelligence_mode", string(resp.ThreatIntelMode))
+
+	if err := d.Set("firewalls", flattenNetworkSubResourceID(resp.Firewalls)); err != nil {
+		return fmt.Errorf(`setting "firewalls": %+v`, err)
+	}
+
+	if err := d.Set("rule_collection_groups", flattenNetworkSubResourceID(resp.RuleGroups)); err != nil {
+		return fmt.Errorf(`setting "rule_collection_groups": %+v`, err)
+	}
+
+	return tags.FlattenAndSet(d, resp.Tags)
+}
+
+func resourceArmFirewallPolicyDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Network.FirewallPoliciesClient
+	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.FirewallPolicyID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	if _, err := client.Delete(ctx, id.ResourceGroup, id.Name); err != nil {
```

Comment:
Is there a reason this isn't doing an async `WaitForCompletionRef` here?

Reply:
Oops, thank you for pointing this out :+1: